### PR TITLE
Update SDK according to `2.0.0-pre-rc.14`

### PR DESCRIPTION
**Pull request by 0x009922**

### CODE REVIEW

File: .changeset/thirty-adults-perform.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+'@iroha2/crypto-target-bundler': minor
+'@iroha2/crypto-target-node': minor
+'@iroha2/crypto-target-web': minor
+'@iroha2/crypto-core': minor
+---
+
+**refactor**: respect new `Algorithm` type & codec from `data-model` and combine it with the crypto's simple `Algorithm` type with is simply a string. Add `Algorithm.toDataModel` and `Algorithm.fromDataModel` methods.
```
Comment (outoftardis) [inline .changeset/thirty-adults-perform.md:8]:
```suggestion
**refactor**: combine new `Algorithm` type and codec from `data-model` with the crypto's `Algorithm` type, which is simply a string. Add `Algorithm.toDataModel` and `Algorithm.fromDataModel` methods.
```

```suggestion
**refactor**: introduce new `Algorithm` type and codec from `data-model`. Add `Algorithm.toDataModel` and `Algorithm.fromDataModel` methods.
```

File: .changeset/large-teachers-boil.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+'@iroha2/data-model': major
+'@iroha2/client': major
+---
+
+Updated according to Iroha `2.0.0-pre-rc.14` (TODO put finaly stable hash here)
+
+Notable changes:
+
+- Structure renamed: ~~`QueryError`~~ `QueryExecutionFailure`
+- `PublicKey` and `PrivateKey` had untyped `digest_function: string` field. Now the new enum struct is introduced - `Algorithm`:
```
[the remark this answers is not in the document]
Comment (0x009922) [reply .changeset/large-teachers-boil.md:12]: I would say:

> Introduced new enum struct, `Algorithm`. `digest_function: string` field in `PublicKey` and `PrivateKey` became `digest_function: Algorithm`:

[the remark this answers is not in the document]
Comment (outoftardis) [reply .changeset/large-teachers-boil.md:12]: then maybe

```suggestion
- Introduced new enum struct, `Algorithm`. Changed `digest_function: string` field in `PublicKey` and `PrivateKey` to `digest_function: Algorithm`:
```

**Review by outoftardis — APPROVED** [review]
docs lgtm

**Review by stefashkaa — APPROVED** [review]

**Pull request merged**